packages/devtools_app/test/integration_tests/logging.dart

```diff
@@ -41,6 +41,35 @@ void loggingTests() {
     expect(await logs.logCount(), greaterThan(0));
   });
 
+  test('filter log data', () async {
+    final DevtoolsManager tools =
+        DevtoolsManager(tabInstance, webdevFixture.baseUri);
+    await tools.start(appFixture);
+    await tools.switchPage('logging');
+
+    final String currentPageId = await tools.currentPageId();
+    expect(currentPageId, 'logging');
+
+    // Cause app to log.
+    final LoggingManager logs = LoggingManager(tools);
+    await logs.clearLogs();
+
+    await appFixture.invoke('controller.emitLog()');
+    await appFixture.invoke('controller.emitLogTest()');
+    await waitFor(() async => await logs.logCount() > 0);
+    final countAll = await logs.logCount();
+
+    await logs.filterLogs('test');
+    final countFilter = await logs.logCount();
+
+    await logs.filterLogs('');
+    final countAllAfter = await logs.logCount();
+
+    // Verify the log data shows up in the UI.
+    expect(countAll, greaterThanOrEqualTo(countFilter));
```

Comment:
using greaterThanEqualTo doesn't verify that the filtering actually did anything
specify exactly how many log messages you expect in these two cases.